main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java

```diff
@@ -0,0 +1,1101 @@
+        bLonSec[0] = ButterKnife.findById(v, R.id.ButtonLonSec_01);
+        bLonPnt[4] = ButterKnife.findById(v, R.id.ButtonLonPnt_10000);
+        bLonPnt[3] = ButterKnife.findById(v, R.id.ButtonLonPnt_01000);
+        bLonPnt[2] = ButterKnife.findById(v, R.id.ButtonLonPnt_00100);
+        bLonPnt[1] = ButterKnife.findById(v, R.id.ButtonLonPnt_00010);
+        bLonPnt[0] = ButterKnife.findById(v, R.id.ButtonLonPnt_00001);
+        tLonLastUnits = ButterKnife.findById(v, R.id.LonLastUnitsChar);
+
+        variablesPanel = ButterKnife.findById(v, R.id.VariablesPanel);
+        variablesScrollableContent = ButterKnife.findById(v, R.id.VariablesScrollpane);
+        variableDivider = ButterKnife.findById(v, R.id.VariableDivider);
+        equationGrid = ButterKnife.findById(v, R.id.EquationTable);
+        variableGrid = ButterKnife.findById(v, R.id.FreeVariableTable);
+
+        tLatResult = ButterKnife.findById(v, R.id.latRes);
+        tLonResult = ButterKnife.findById(v, R.id.lonRes);
+
+        notes = ButterKnife.findById(v, R.id.notes_text);
+
+        inputLatHem = gp.getLatitude() > 0 ? "N" : "S";
+        bLatHem.setText(inputLatHem);
+        setCoordValue(gp.getLatDeg(), bLatDeg[0], bLatDeg[1], bLatDeg[2]);
+        setCoordValue(gp.getLatMin(), bLatMin[0], bLatMin[1]);
+        setCoordValue(gp.getLatSec(), bLatSec[0], bLatSec[1]);
+
+        inputLonHem = gp.getLatitude() > 0 ? "E" : "W";
+        bLatHem.setText(inputLonHem);
+        setCoordValue(gp.getLonDeg(), bLonDeg[0], bLonDeg[1], bLonDeg[2]);
+        setCoordValue(gp.getLonMin(), bLonMin[0], bLonMin[1]);
+        setCoordValue(gp.getLonSec(), bLonSec[0], bLonSec[1]);
+
+        latButtons = Arrays.asList(bLatDeg[2], bLatDeg[1], bLatDeg[0],
+                                               bLatMin[1], bLatMin[0],
+                                               bLatSec[1], bLatSec[0],
+           bLatPnt[4], bLatPnt[3], bLatPnt[2], bLatPnt[1], bLatPnt[0]);
+
+        lonButtons = Arrays.asList(bLonDeg[2], bLonDeg[1], bLonDeg[0],
+                                               bLonMin[1], bLonMin[0],
+                                               bLonSec[1], bLonSec[0],
+           bLonPnt[4], bLonPnt[3], bLonPnt[2], bLonPnt[1], bLonPnt[0]);
+
+        coordButtons = new ArrayList<>(latButtons.size() + lonButtons.size());
+        coordButtons.addAll(latButtons);
+        coordButtons.addAll(lonButtons);
+
+        minButtons = Arrays.asList(tLatDegChar, bLatMin[1], bLatMin[0],
+                                   tLonDegChar, bLonMin[1], bLonMin[0]);
+        secButtons = Arrays.asList(tLatMinChar, bLatSec[1], bLatSec[0],
+                                   tLonMinChar, bLonSec[1], bLonSec[0]);
+        pointLowButtons = Arrays.asList(bLatPnt[4], bLatPnt[3],
+                                        bLonPnt[4], bLonPnt[3]);
+        lastUnits = Arrays.asList(tLatLastUnits, tLonLastUnits);
+
+        bLatDeg[2].setNextButton(bLatDeg[1]);

```

Comment:
There are similar things done twice: once for the lat buttons and another time for the lon buttons. Maybe this can be extracted into a method which is called twice.